File: Data_Structures/javascript/BinaryTree/postOrderTraversal.js
const PostOrderTraverse = (node) => {
   if (node !== null) {
     //Traverse the left subtree
     PostOrderTraverse(node.left);

     //Traverse the right subtree
     PostOrderTraverse(node.right);

     //Print the node
     console.log(node.key);
   }
};

const tree = new BinarySearchTree();
tree.insert(11);
tree.insert(9);
tree.insert(15);
tree.insert(8);
tree.insert(10);
tree.insert(12);
tree.insert(16);

const root = tree.getRoot();
PostOrderTraverse(root);

File: Data_Structures/C++/BinarySearchTree/postOrderTraversal.cpp
#include<iostream>

using namespace std;

struct Node {
    char data;
    struct Node *left;
    struct Node *right;
};

//Function to visit nodes in Postorder
void Postorder(Node *root) {
    if(root == NULL) return;

    Postorder(root->left);    // Visit left subtree
    Postorder(root->right);   // Visit right subtree
    printf("%c ",root->data); // Print data
}

// Function to Insert Node in a Binary Search Tree
Node* Insert(Node *root,char data) {
    if(root == NULL) {
	root = new Node();
	root->data = data;
	root->left = root->right = NULL;
    }
    else if(data <= root->data)
	root->left = Insert(root->left,data);
    else
	root->right = Insert(root->right,data);
    return root;
}

int main() {
/*
 * Code To Test the logic Creating an example tree
 *		M
 *	       / \
 *	      B   Q
 *	     / \   \
 *	    A   C   Z
 */
    Node* root = NULL;
    root = Insert(root,'M'); root = Insert(root,'B');
    root = Insert(root,'Q'); root = Insert(root,'Z');
    root = Insert(root,'A'); root = Insert(root,'C');
    cout<<"Postorder: ";
    Postorder(root);
    cout<<"\n";
}
